src/preact/hooks/useLocalStorage.js

```diff
@@ -0,0 +1,37 @@
+import {useCallback, useRef, useState} from '#preact';
+
+/**
+ * @param {string} key
+ * @param {any} [defaultValue]
+ * @return {{ "0": any, "1": function(any): void }}
+ */
+export function useLocalStorage(key, defaultValue = null) {
+  // Keep track of the state locally:
+  const [value, setValue] = useState(() => {
+    try {
+      // eslint-disable-next-line local/no-forbidden-terms
+      const json = self.localStorage?.getItem(key);
+      return !json ? defaultValue : JSON.parse(json);
+    } catch (err) {
+      // warning: could not read from local storage
+      return defaultValue;
+    }
+  });
+
+  // Use a keyRef to ensure that `storeValue` maintains referential integrity
+  const keyRef = useRef(key);
+  keyRef.current = key;
```

Comment:
Nit: This could cause some weird behavior if the `key` were changed after initialization. I understand we're using `keyRef` so that the `useCallback` doesn't warn about the missed memo dependency, but I don't think we should ever update the value after the initial set.

Reply:
Yeah, that's true ... I removed this `useRef` optimization, and added `key` as a dependency to the `useCallback` so that we don't have to worry about unexpected behavior.